### PR TITLE
Master writing and more

### PR DESCRIPTION
**Pull request by profxj**
Started out as adding the output of MasterFrames when the
code exits.    

Evolved into a series of fixes realted to Python 3. 
And some effort to calibrate the R400 grating on LRISr

### CODE REVIEW

File: bin/run_pypit [hunk 1]
```diff
@@ -57,10 +58,11 @@ def main() :
 
     # Load options from command line
     try:
-        opt, arg = getopt.getopt(sys.argv[1:], 'hmqc:v:', ['help',
+        opt, arg = getopt.getopt(sys.argv[1:], 'hmqcd:v:', ['help',
                                                            'use_masters',
                                                           'quick',
                                                           'cpus',
+                                                            'develop',
```
Comment (rcooke-ast) [inline bin/run_pypit]: Good idea!  You'll also need to include a description of this command line option on line 117(ish) of armsgs.py

File: bin/run_pypit [hunk 2]
```diff
@@ -83,11 +88,11 @@ def main() :
 
     # Execute the reduction, and catch any bugs for printout
     if debug['develop']:
-        pypit.PYPIT(red, progname=pypit.__file__, quick=qck, ncpus=cpu, verbose=vrb,
+        pypit.PYPIT(red, initmsgs, progname=pypit.__file__, quick=qck, ncpus=cpu, verbose=vrb,
```
Comment (rcooke-ast) [inline bin/run_pypit]: As I comment below in PYPIT, I think we should not pass initmsgs here. Instead, we should pass a develop keyword (like what is presently done with the verbose keyword), or perhaps better, a dictionary containing some of the relevant bits of information for the messages (including `verbose, develop, last_updated, version`)

File: pypit/arlris.py [hunk 3]
```diff
@@ -97,22 +99,22 @@ def read_lris(raw_file, det=None, TRIM=False):
...
+    postpix = postpix // xbin
 
     # Deal with detectors
     if det in [1,2]:
-        nx = nx / 2
-        n_ext = n_ext / 2
-        det_idx = np.arange(n_ext) + (det-1)*n_ext
+        nx = nx // 2
+        n_ext = n_ext // 2
+        det_idx = np.arange(n_ext).astype(int) + (det-1)*n_ext
```
Comment (rcooke-ast) [inline pypit/arlris.py]: For line 113 and 117, we can use (I think) `np.arange(n_ext, dtype=np.int)` to save recasting

File: pypit/arutils.py [hunk 4]
```diff
@@ -146,10 +147,12 @@ def dummy_self(pypitdir=None):
     # Dummy Class
     slf = type('Dummy', (object,), {"_argflag": {}, "_spect": {}})
     slf._argflag['run'] = {}
-    if pypitdir is not None:
+    if pypitdir is None:
+        pypitdir = __file__[0:__file__.rfind('/')]
         slf._argflag['run']['pypitdir'] = pypitdir
```
Comment (rcooke-ast) [inline pypit/arutils.py]: I think this line (152) should be outside the if statement...

File: pypit/pypit.py [hunk 5]
```diff
@@ -8,19 +10,22 @@
...
 import traceback
+from pypit import armsgs
 
 # Import PYPIT routines
+'''
 import ardebug
 debug = ardebug.init()
-#debug['develop'] = True
-#debug['arc'] = True
+debug['develop'] = True
```
Comment (rcooke-ast) [inline pypit/pypit.py]: Oops!  ;-)

File: pypit/pypit.py [hunk 6]
```diff
@@ -67,9 +72,10 @@ def PYPIT(redname, progname=__file__, quick=False, ncpus=1, verbose=1,
     ---------------------------------------------------
     """
     # Init logger
-    import armsgs
-    msgs = armsgs.get_logger((logname, debug, last_updated, version, verbose))
-    import arload
+    msgs = armsgs.get_logger((logname, imsgs._debug, imsgs._last_updated,
```
Comment (rcooke-ast) [inline pypit/pypit.py]: I'm not sure imsgs is used after this line, and we only use some of the variables (i.e. not the class itself). So, I think it's best not to pass imsgs as an argument into PYPIT

File: bin/run_pypit (hunk 1 above)
Comment (profxj) [inline bin/run_pypit]: Done.  I should note that there is something I don't understand about the parsing.
Am raising an issue..

File: bin/run_pypit (hunk 2 above)
Comment (profxj) [inline bin/run_pypit]: A dictionary it shall be.

Comment (profxj) [inline bin/run_pypit]: Upon further review, I've done some mixing about.  
Didn't make sense to me to pass around version and last_updated.

File: pypit/arlris.py (hunk 3 above)
Comment (profxj) [inline pypit/arlris.py]: Done

File: pypit/arutils.py (hunk 4 above)
Comment (profxj) [inline pypit/arutils.py]: Good catch

File: pypit/pypit.py (hunk 5 above)
Comment (profxj) [inline pypit/pypit.py]: wiped out. :)

File: pypit/pypit.py (hunk 6 above)
Comment (profxj) [inline pypit/pypit.py]: Agreed.  Modified

**Comment by rcooke-ast** [comment]
Looks good to me... merging.

**Pull request merged**